butterknife-compiler/src/main/java/butterknife/compiler/ButterKnifeProcessor.java

```diff
@@ -1143,17 +1147,19 @@ private String getPackageName(TypeElement type) {
     return elementUtils.getPackageOf(type).getQualifiedName().toString();
   }
 
-  private static boolean hasAnnotationWithName(Element element, String simpleName) {
+  private static boolean hasAnnotationWithNames(Element element, String... simpleNames) {
     for (AnnotationMirror mirror : element.getAnnotationMirrors()) {
       String annotationName = mirror.getAnnotationType().asElement().getSimpleName().toString();
-      if (simpleName.equals(annotationName)) {
-        return true;
+      for (String simpleName : simpleNames) {
+        if (simpleName.equals(annotationName)) {
+          return true;
+        }
       }
     }
     return false;
   }
 
   private static boolean isRequiredBinding(Element element) {
-    return !hasAnnotationWithName(element, NULLABLE_ANNOTATION_NAME);
+    return !hasAnnotationWithNames(element, NULLABLE_ANNOTATION_NAME, OPTIONAL_ANNOTATION_NAME);
```

Comment:
So should it just do a manual check here as well to make sure they're valid? Or separate `isOptionalBinding` method (and all that comes with that) entirely? 

I was living it to the annotations themselves, but not actually sure if the compiler will yell at you for using, say, a `@Target(METHOD)` annotation on a field or vice versa.